core/tauri/src/api/process/command.rs

```diff
@@ -390,6 +406,50 @@ impl Command {
   }
 }
 
+// adapted from https://doc.rust-lang.org/std/io/trait.BufRead.html#method.read_line
+fn read_command_output<R: BufRead + ?Sized>(
+  r: &mut R,
+  buf: &mut Vec<u8>,
+) -> std::io::Result<usize> {
+  let mut read = 0;
+  loop {
+    let (done, used) = {
+      let available = match r.fill_buf() {
+        Ok(n) => n,
+        Err(ref e) if e.kind() == std::io::ErrorKind::Interrupted => continue,
+        Err(e) => return Err(e),
+      };
+      match memchr::memchr(b'\n', available) {
+        Some(i) => {
+          buf.extend_from_slice(&available[..=i]);
+          (true, i + 1)
```

Comment:
this is a micro optimization because exclusive ranges still seem be faster (not that inclusive isn't also fast), but we may as well do it since we calculate it anyways.
```suggestion
          let end = i + 1;
          buf.extend_from_slice(&available[..end]);
          (true, end)
```